app/src/main/java/com/macoscope/ketchuplunch/presenter/WeeksPresenter.kt

```diff
@@ -0,0 +1,62 @@
+package com.macoscope.ketchuplunch.presenter
+
+import android.app.Activity
+import com.google.api.client.googleapis.extensions.android.gms.auth.UserRecoverableAuthIOException
+import com.macoscope.ketchuplunch.model.lunch.Week
+import com.macoscope.ketchuplunch.model.lunch.WeeksService
+import com.macoscope.ketchuplunch.view.lunch.WeeksView
+import org.jetbrains.anko.AnkoLogger
+import org.jetbrains.anko.error
+import rx.Observable
+import rx.android.schedulers.AndroidSchedulers
+import rx.lang.kotlin.deferredObservable
+import rx.lang.kotlin.plusAssign
+import rx.lang.kotlin.subscriber
+import rx.schedulers.Schedulers
+import rx.subscriptions.CompositeSubscription
+
+class WeeksPresenter(val weeksService: WeeksService, val weeksView: WeeksView) : AnkoLogger {
+    val subscriptions: CompositeSubscription = CompositeSubscription()
+    private val REQUEST_AUTHORIZATION = 1001
+    private fun loadData() {
+        weeksView.showLoading()
+        subscriptions += loadWeeksObservable().subscribe(
+                subscriber<List<Week>>()
+                        .onNext {
+                            weeksView.showWeeks(it)
+                            weeksView.hideLoading()
```

Comment:
Create private method `showWeeks` and move there weeksView.showWeeks(it) and weeksView.hideLoading()